### ISSUE TITLE
NoTorsionError in test_geometry.py

### ISSUE THREAD

**Issue by jchodera** [choderalab/perses#310]
When running `perses/tests/test_geometry.py` locally, our old friend `no eligible torsions found` is back:
```
======================================================================
ERROR: Try proposing geometries for all T4 ligands from all T4 ligands
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/choderaj/miniconda3/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/Users/choderaj/github/choderalab/perses/perses.choderalab/perses/tests/test_geometry_engine.py", line 1020, in test_propose_lysozyme_ligands
    run_proposals(proposals)
  File "/Users/choderaj/github/choderalab/perses/perses.choderalab/perses/tests/test_geometry_engine.py", line 1051, in run_proposals
    new_positions, logp = geometry_engine.propose(top_proposal, current_positions, beta)
  File "/Users/choderaj/github/choderalab/perses/perses.choderalab/perses/rjmc/geometry.py", line 121, in propose
    logp_proposal, new_positions = self._logp_propose(top_proposal, current_positions, beta, direction='forward')
  File "/Users/choderaj/github/choderalab/perses/perses.choderalab/perses/rjmc/geometry.py", line 204, in _logp_propose
    atom_proposal_order, logp_choice = proposal_order_tool.determine_proposal_order(direction='forward')
  File "/Users/choderaj/github/choderalab/perses/perses.choderalab/perses/rjmc/geometry.py", line 1936, in determine_proposal_order
    logp_torsion_choice += add_atoms(new_heavy_atoms, atoms_torsions)
  File "/Users/choderaj/github/choderalab/perses/perses.choderalab/perses/rjmc/geometry.py", line 1925, in add_atoms
    chosen_torsion, logp_choice = self._choose_torsion(atoms_with_positions, atom)
  File "/Users/choderaj/github/choderalab/perses/perses.choderalab/perses/rjmc/geometry.py", line 1981, in _choose_torsion
    raise NoTorsionError("No eligible torsions found for placing atom %s." % str(atom_for_proposal))
perses.rjmc.geometry.NoTorsionError: No eligible torsions found for placing atom <Atom C5 [4]; In MOL_0 0>.
-------------------- >> begin captured logging << --------------------
```

**Comment by jchodera** [comment]
Maybe the [`testsystem-t4`](https://github.com/choderalab/perses/tree/testsystem-t4) branch needs to be merged in to fix this?

**Comment by pgrinaway** [comment]
Ah, I know what this is. I'll be back in on the 3rd to fix this. It would be a lot easier to explain on the whiteboard, but the short of it is that the way that molecules are currently being proposed involves a chance of proposing something un-proposable. It's definitely a bug*, and as I said I'll fix this as soon as I'm back.

* This doesn't affect the T4 set because to generate that I actually tried to generate `ProposalOrders` for each pair, so my script for that actually caught these. So the T4 set there is clean, but the proposals need to be better restricted.

**Comment by jchodera** [comment]
Got it, thanks!

**Comment by pgrinaway** [comment]
Resolved in #313
